### PR TITLE
[WB-8211] Pin deepchem to <2.6.1 to avoid pip conflict

### PR DESCRIPTION
**Pull request by kptkin**
Related to this issue: [WB-8211](https://wandb.atlassian.net/browse/WB-8211)

Description
-----------
What does the PR do?
Deepchem had a new release (2.6.1) that causes conflicts with pip install. 
This PR is pinning to the latest release version that was stable (2.5.0)

Testing
-------
How was this PR tested?
Fixes existing test.

Checklist
-------
- Name PR "[WB-NNNN][WB-MMMM] Add support for..." similar to entries in CHANGELOG.md
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -54,14 +54,14 @@ commands:
             - save_cache:
                 paths:
                   - ./.tox
-                key: v0.28-toxenv-{{ .Environment.CIRCLE_BRANCH }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "tox.ini" }}-{{ checksum "setup.py" }}-{{ checksum "requirements.txt" }}-{{ checksum "requirements_dev.txt" }}
+                key: v0.29-toxenv-{{ .Environment.CIRCLE_BRANCH }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "tox.ini" }}-{{ checksum "setup.py" }}-{{ checksum "requirements.txt" }}-{{ checksum "requirements_dev.txt" }}
```
Comment (dmitryduev) [inline .circleci/config.yml:57]: Could you please educate me, when should this be ++'d?

**Review by dmitryduev — APPROVED** [review]
🍾  thanks a bunch, @kptkin!!

**Comment by dmitryduev** [comment]
BTW, how do we know that `func-s_imports-lin-py37` didn't use cache like it did on master?

File: .circleci/config.yml (hunk 1 above)
Comment (kptkin) [reply .circleci/config.yml:57]: Ah good question, I did it now to force it not to use the cache. I don't think we have a good protocol about how we update it. One thing would be obviously if we change requirements, or config changes. I think this is a balance between lowering our cost/time with circleci and hitting case as this one that you discovered.

@raubitsj could probably add more context :)

**Comment by kptkin** [comment]
> BTW, how do we know that `func-s_imports-lin-py37` didn't use cache like it did on master?

image

https://app.circleci.com/pipelines/github/wandb/client/9867/workflows/df8f240b-aaa9-4987-9cb0-47e0ac280935/jobs/107320

File: .circleci/config.yml (hunk 1 above)
Comment (dmitryduev) [reply .circleci/config.yml:57]: Got it, thanks. Definitely deserves a section in CONTRIBUTING.md :)
Can't wait to get this merged in, it'll make my Settings PR unbelievable green!

**Pull request merged**